modules/apps/style-book/style-book-test/src/testIntegration/java/com/liferay/style/book/service/test/StyleBookEntryServiceTest.java

```diff
@@ -121,6 +123,47 @@ public void testDeleteStyleBookEntryByExternalReferenceCodeWithoutDeletePermissi
 		}
 	}
 
+	@Test
+	public void testFetchStyleBookEntryByExternalReferenceCode()
+		throws Exception {
+
+		StyleBookEntry styleBookEntry =
+			_styleBookEntryService.addStyleBookEntry(
+				RandomTestUtil.randomString(), _group.getGroupId(),
+				RandomTestUtil.randomString(), null,
+				RandomTestUtil.randomString(), _serviceContext);
+
+		StyleBookEntry curStyleBookEntry =
+			_styleBookEntryService.fetchStyleBookEntryByExternalReferenceCode(
+				styleBookEntry.getExternalReferenceCode(),
+				styleBookEntry.getGroupId());
+
+		Assert.assertEquals(
+			styleBookEntry.getStyleBookEntryId(),
+			curStyleBookEntry.getStyleBookEntryId());
+
+		_removeResourcePermissions(styleBookEntry.getStyleBookEntryId());
+
+		try {
+			UserTestUtil.setUser(
+				UserTestUtil.addGroupUser(_group, RoleConstants.SITE_MEMBER));
+
+			_styleBookEntryService.fetchStyleBookEntryByExternalReferenceCode(
+				styleBookEntry.getExternalReferenceCode(),
+				styleBookEntry.getGroupId());
+
+			Assert.fail();
+		}
+		catch (PrincipalException principalException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(principalException);
+			}
+		}
+		finally {
+			UserTestUtil.setUser(TestPropsValues.getUser());
+		}
```

Comment:
I think you can create another test for the permission part :thinking: like the `testGetStyleBookEntryByExternalReferenceCode` one is doing

Reply:
Oh, you did in the next commit. Try to re-order and keep the performance improvement commit as the first one for the tests part, so it is already clear about this part :slightly_smiling_face: